src/store/modules/DownloadModule.ts

```diff
@@ -70,99 +71,86 @@ export const DownloadModule = {
-            const profileModList = throwForR2Error(await ProfileModList.getModList(params.profile));
-
-            const modAlreadyInstalled = profileModList.find(
-                value => value.getName() === params.combo.getMod().getFullName()
-                    && value.getVersionNumber().isEqualTo(params.combo.getVersion().getVersionNumber())
-            );
-
-            if (modAlreadyInstalled !== undefined && modAlreadyInstalled) {
-                return;
-            }
-
-            const manifestMod = new ManifestV2().fromThunderstoreCombo(params.combo);
-            const olderInstallOfMod = profileModList.find(value => value.getName() === manifestMod.getName());
-
-            throwForR2Error(await ProfileInstallerProvider.instance.uninstallMod(manifestMod, params.profile));
-            throwForR2Error(await ProfileInstallerProvider.instance.installMod(manifestMod, params.profile));
-            throwForR2Error(await ProfileModList.addMod(manifestMod, params.profile));
-
-            if (olderInstallOfMod === undefined || olderInstallOfMod.isEnabled()) {
-                return;
-            }
-
-            await ProfileModList.updateMod(manifestMod, params.profile, async mod => {
-                mod.disable();
-            });
-            await ProfileInstallerProvider.instance.disableMod(manifestMod, params.profile);
-        },
-
-        async installMods({commit, dispatch}, params: {
-            downloadedMods: ThunderstoreCombo[],
-            downloadId: UUID,
+        async downloadAndInstallCombos({commit, dispatch}, params: {
+            combos: ThunderstoreCombo[],
             profile: ImmutableProfile,
+            game: Game
         }) {
-            await ProfileModList.requestLock(async () => {
-                let currentDownloadIndex = 0;
-                for (const combo of params.downloadedMods) {
-                    try {
-                        await dispatch('installMod', {profile: params.profile, combo});
-                    } catch (e) {
-                        throw R2Error.fromThrownValue(e, `Failed to install mod [${combo.getMod().getFullName()}]`);
-                    }
-                    commit('updateDownload', {
-                        downloadId: params.downloadId,
-                        modName: combo.getMod().getName(),
-                        installProgress: DownloadUtils.generateProgressPercentage(100, currentDownloadIndex, params.downloadedMods.length)
-                    });
-                    currentDownloadIndex++;
-                }
+            const { combos, game, profile } = params;
 
-                const modList = throwForR2Error(await ProfileModList.getModList(params.profile));
-                throwForR2Error(await ConflictManagementProvider.instance.resolveConflicts(modList, params.profile));
-            });
+            const downloadId = await dispatch('_addDownload', combos);
```

Comment:
Minor: perhaps this too should be in the `try` block so it gets error handled like the rest? That would mean the catch-block needs to check if `downloadId` is set, but that sound like a fair tradeoff.